### PR TITLE
Adding docs copied from microk8s and adjusted for k8s

### PR DESCRIPTION
**Pull request by addyess**
## Overview

Adds documentation to use velero and minio (though it could be some other s3 bucket endpoint) to facilitate cluster backups

### Details
* Based on the [Microk8s docs](https://microk8s.io/docs/velero) under the same banner, these docs offer a tutorial of how to install velero on minio in order to create backups.  
* adjusted those docs to upgrade the versions of velero and minio used in the examples.

### CODE REVIEW

File: docs/src/snap/tutorial/backing-up.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+```bash
+wget https://github.com/vmware-tanzu/velero/releases/download/v1.14.0/velero-v1.14.0-linux-amd64.tar.gz 
+tar -xzf velero-v1.14.0-linux-amd64.tar.gz
+chmod +x velero-v1.14.0-linux-amd64/velero
+sudo chown root:root velero-v1.14.0-linux-amd64/velero
+sudo mv velero-v1.14.0-linux-amd64/velero /usr/local/bin/velero
+```
+
+Before installing Velero, we export the kubeconfig file from MicroK8s.
```
Comment (eaudetcobello) [inline docs/src/snap/tutorial/backing-up.md:81]:
```suggestion
Before installing Velero, we export the kubeconfig file from using the `config` command.
```
Leftover microk8s ref.

File: docs/src/snap/tutorial/backing-up.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+```bash
+mkdir -p $HOME/.kube
+sudo k8s kubectl config view --raw > $HOME/.kube/config
+```
+
+We also export the MinIO credentials so we can feed them to Velero.
```
Comment (eaudetcobello) [inline docs/src/snap/tutorial/backing-up.md:88]:
```suggestion
We also export the MinIO credentials so we can provide them to Velero.
```
Feed sounds... vampiric.

File: docs/src/snap/tutorial/backing-up.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+SERVICE=$(sudo k8s kubectl -n velero get svc -l app=minio -o jsonpath="{.items[0].metadata.name}")
+cat <<EOF > credentials-velero
+[default]
+    aws_access_key_id=${ACCESS_KEY}
+    aws_secret_access_key=${SECRET_KEY}
+EOF
+```
+
+We are now ready to install Velero, with the aws [plugin
+matching][aws-plugin-matching] the velero release:
```
Comment (eaudetcobello) [inline docs/src/snap/tutorial/backing-up.md:102]: I think the link is a bit wonky here.

File: docs/src/snap/tutorial/backing-up.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+## Backup workloads
+
+To backup the `workloads` namespace we use the `--include-namespaces` argument:
+ 
+```bash
+velero backup create workloads-backup --include-namespaces=workloads
+```
+
+> ⓘ **Note:** Please, consult the [official Velero
```
Comment (eaudetcobello) [inline docs/src/snap/tutorial/backing-up.md:126]:
```suggestion
> **Note:** Please, consult the [official Velero
```

I like the aesthetic but let's remove it to keep consistent with the rest of the docs here. (Example: https://github.com/canonical/k8s-snap/blob/f0130a9f2399367a6e4a18457737da0a3a6cca4c/docs/src/snap/tutorial/kubectl.md?plain=1#L32)

**Review by eaudetcobello — CHANGES_REQUESTED** [review]
Nice doc!

File: docs/src/snap/howto/backup-restore.md [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+```bash
+sudo k8s enable local-storage
+```
+
+Helm is used to setup [MinIO][MinIO Charts] under the `velero` namespace:
+
+```bash
+sudo k8s helm repo add minio https://charts.min.io
+sudo k8s kubectl create namespace velero
+
```
Comment (evilnick) [inline docs/src/snap/howto/backup-restore.md:46]: Can we just pick one here?

File: docs/src/snap/howto/backup-restore.md [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+```bash
+wget https://github.com/vmware-tanzu/velero/releases/download/v1.14.0/velero-v1.14.0-linux-amd64.tar.gz 
+tar -xzf velero-v1.14.0-linux-amd64.tar.gz
+chmod +x velero-v1.14.0-linux-amd64/velero
+sudo chown root:root velero-v1.14.0-linux-amd64/velero
+sudo mv velero-v1.14.0-linux-amd64/velero /usr/local/bin/velero
+```
+
+Before installing Velero, we export the kubeconfig file using the `config`
```
Comment (evilnick) [inline docs/src/snap/howto/backup-restore.md:88]: Maybe this step should go first then?

File: docs/src/snap/howto/backup-restore.md [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+This document describes how to setup Velero with the MinIO provider acting
+as an S3 compatible object store.
+
+
+## What you will need
+
+- A running Canonical Kubernetes with DNS enabled
+- MinIO (install described below)
+- Velero (install described below))
+- An example workload
```
Comment (evilnick) [inline docs/src/snap/howto/backup-restore.md:21]: - Credentials for AWS or another supported cloud provider ?

**Comment by evilnick** [comment]
Nice job @addyess. I moved it to the howto section and fixed up some other bits - a few comments for your perusal

File: docs/src/snap/howto/backup-restore.md (hunk 5 above)
Comment (addyess) [reply docs/src/snap/howto/backup-restore.md:45]: COOL... picked

File: docs/src/snap/howto/backup-restore.md (hunk 6 above)
Comment (addyess) [reply docs/src/snap/howto/backup-restore.md]: I think i needed to reword this a bit.  The steps were right. 

there's "install a tool" and "install into the cluster"

File: docs/src/snap/howto/backup-restore.md (hunk 7 above)
Comment (addyess) [reply docs/src/snap/howto/backup-restore.md:20]: I don't think AWS is required -- just that MinIO + Velero uses credentials that LOOK like AWS creds.  Really its just S3 creds.  I've updated the docs to make that more clear

**Pull request merged**